druid/src/widget/list.rs

```diff
@@ -195,27 +211,40 @@ impl<S: Data, T: Data> ListIter<(S, T)> for (S, Arc<Vec<T>>) {
 
 impl<C: Data, T: ListIter<C>> Widget<T> for List<C> {
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut T, env: &Env) {
-        let mut children = self.children.iter_mut();
-        data.for_each_mut(|child_data, _| {
-            if let Some(child) = children.next() {
-                child.event(ctx, event, child_data, env);
-            }
-        });
+        if !self.scroll_component.filter_event(ctx, event, env) {
```

Comment:
Maybe using `set_handled` and `is_handled` would make sense? Somewhat like this:
```rust
self.scroll_component.event(ctx, event, env);
if !ctx.is_handled() {
    do_your_handling();
}
```

Reply:
That is true, it would definitely be more idiomatic when compared to other similar situations. It could also consume the event and return `Option<Event>` thus requiring an `if let Some(event) = filter_event(event) {}` but I'll be the first to admit that that feels heavy handed especially when the goal of these changes is to let widgets which need to be able to break out of the usual scrolling logic when handling atypical situations.

Reply:
Event filtering now uses `ctx.set_handled()` for the sake of being idiomatic